app/components/oral_history/downloads_list_component.rb

```diff
@@ -24,7 +24,8 @@ def initialize(work:, decorator:)
     def all_members
       @all_members ||= begin
         members = work.members.includes(:leaf_representative)
-        members = members.where(published: true) if current_user.nil?
+        show_all_members = (can? :read, Asset) && (can? :read, Work)
+        members = members.where(published: true) unless show_all_members
```

Comment:
I think avoiding the temporary variable is a lot more clear:

```
unless (can? :read, Asset) && (can? :read, Work)
   members = members.where(published: true) unless show_all_members
end
```

Although yeah, i guess that condition is a bit confusing too, I don't totally understand how access_granted works hwen you pass that class instead of an isntance... or why we need/want to check both...    For the second question, should we just replace it everywhere with `can? :read, Kithe::Model` actually?  

But for the first question... `can :read, ModelClass` is magically true only of those accounts that have the role where we say they can read regardless of published status?  

Hmm, now I'm wondering... maybe for this one we _do_ want to make a helper method in `ApplciationHelper`, like `can_see_unpublished_records?`, since it is a bit weird/unclear, document it there, and delegate to it from the view components? I dunno if that improves things or not, just kind of writing as I think.

Reply:
Unfortunately, AccessGranted is not smart enough to deduce that e.g. `can? :destroy, Kithe::Model` in the access_policy file implies `can? :destroy, Work`.

I also found your argument in https://github.com/sciencehistory/scihist_digicoll/pull/1976#issuecomment-1334473404 convincing . I find using the basic can? syntax in the various parts of the code where it's needed is actually helpful in this part of the process. So I'm keeping everything simple -- if a little verbose -- as of this refactor. 

My feeling here is that pnce we actually get the various permissions changes sorted out and tested, we can of course explore more concise and elegant formulations in a separate PR.

Reply:
OK, so:

a) I still think factoring this confusing check out into re-usable helper is a good idea either way, but

b) I want to play with access_granted a bit more.  
   * i believe you that if the policy says `can :read, Kithe::Model`, then checking `can? :read, Work` won't succesfully return true -- but I want to investigate this more and see if we can submit a PR to fix it in access_granted
   * But even notwithstanding -- okay, butr what if we do `can :read, Kithe::Model` in the policy, but just _check_ `can? :read, Kithe::Model` instead? That should work. And I think right now every single place does `can?(:read, Work) && can?(:read, Asset)` anyway -- it seems _preferable_ to say `can? :read, Kithe::Model` there _anyway_, plus put `can :read, Kithe::Model` in the policy.

Reply:
(Summary -- this is what we ended up doing.)